torch/distributed/device_mesh.py

```diff
@@ -89,6 +89,7 @@ def create_child_mesh(
                     device_mesh.device_type,
                     mesh_1d,
                     mesh_dim_names=(mesh_dim_name,),
+                    _init_backend=False,
```

Comment:
@wz337 Even though `_init_backend` is defaulted to `True`, here we are passing it as `False` for child meshes. For FSDP2 2D training, I was thinking that this means the child meshes are now missing the `self._coordinate_on_dim` and `self._dim_group_infos`.

Reply:
there seems to have some test failures, so trying to fix it directly in this PR

Reply:
tested locally should fixed those tests

Reply:
How about `_self._dim_group_infos`? I think we need that to support `.get_group()`.

Reply:
those are already populated when we create the submesh?https://github.com/pytorch/pytorch/blob/main/torch/distributed/device_mesh.py#L96

Reply:
Ah, got it! Makes sense!

Reply:
basically when use `init_backend=False` I think there would be some manual pg creation or pg reuse, so we would need to explicitly attach the dim_group_info after `DeviceMesh` constructor?

i.e. for things like `from_group`, inside this API after `DeviceMesh` constructed, we need to attach the dim_group_info to the DeviceMesh created, the `dim_group_info` need to be generated in a similar way as https://github.com/pytorch/pytorch/blob/main/torch/distributed/device_mesh.py#L314-L319